hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RegionScannerImpl.java

```diff
@@ -240,6 +240,9 @@ public synchronized boolean next(List<Cell> outResults, ScannerContext scannerCo
     }
     region.startRegionOperation(Operation.SCAN);
     try {
+      if (scannerContext == null) {
```

Comment:
I guess the remaining implication of this would be we are now creating 2 ScannerContext for every Get instead of 1. So I'll just keep the behavior of only creating a ScannerContext if necessary.

Closing this PR and submitting a follow-up, since the original was reverted.

Reply:
I forgot to submit comment from earlier this mornings... see above @bbeaudreault  Thanks.